nilearn/plotting/displays/_figures.py

```diff
@@ -104,3 +130,355 @@ def savefig(self, output_file=None):
+             (n_vertices, s) x,y,z coordinates of vertices that trace region
+             of interest.
+
+        Notes
+        -----
+        For each face on the edge of a region
+            1. Get a centroid for each face (parallel to the triangle plane)
+            2. Find the xyz coordinate that is normal to the triangle face
+                (at distance `elevation`)
+            3. Arrange the centroids such in a good order for plotting
+        """
+        # Mask indicating faces whose centroids will compose the boundary.
+        edge_faces = self._get_faces_on_edge(parc_idx=parc_idx)
+
+        # gather the centroids of each face
+        centroids = []
+        segments = []
+        vs = []
+        idxs = []
+        for e, face in zip(edge_faces, self._faces):
+            if e:
+                t0 = self._coords[face[0]]
+                t1 = self._coords[face[1]]
+                t2 = self._coords[face[2]]
+
+                # the xyz coordinate is weighted toward the roi boundary (2:1)
+                w0 = 2 if face[0] in parc_idx else 1
+                w1 = 2 if face[1] in parc_idx else 1
+                w2 = 2 if face[2] in parc_idx else 1
+                x = np.average((t0[0], t1[0], t2[0]), weights=(w0, w1, w2))
+                y = np.average((t0[1], t1[1], t2[1]), weights=(w0, w1, w2))
+                z = np.average((t0[2], t1[2], t2[2]), weights=(w0, w1, w2))
+                centroids.append(
+                    self._project_above_face(
+                        np.array((x, y, z)), t0, t1, t2, elevation=elevation
+                    )
+                )
+                segs = [None] * 3
+                if face[0] in parc_idx and face[1] in parc_idx:
+                    segs[0] = self._transform_coord_to_plane(
+                        t0, t0, t1, t2
+                    ) + self._transform_coord_to_plane(t1, t0, t1, t2)
+                if face[0] in parc_idx and face[2] in parc_idx:
+                    segs[1] = self._transform_coord_to_plane(
+                        t0, t0, t1, t2
+                    ) + self._transform_coord_to_plane(t2, t0, t1, t2)
+                if face[1] in parc_idx and face[2] in parc_idx:
+                    segs[2] = self._transform_coord_to_plane(
+                        t2, t0, t1, t2
+                    ) + self._transform_coord_to_plane(t1, t0, t1, t2)
+                segments.append(tuple(segs))
+                vs.append((t0, t1, t2))
+                idxs.append([f for f in face if f in parc_idx])
+
+        centroids = np.array(centroids)
+
+        # Next, sort centroids along boundary
+        # Start with the first vertex
+        current_vertex = 0
+        visited_vertices = {current_vertex}
+        last_distance = np.inf
+        prev_first = 0
+
+        sorted_vertices = [centroids[0]]
+
+        # Loop over the remaining vertices in order of distance from the
+        # current vertex
+        while len(visited_vertices) < len(centroids):
```

Comment:
Can we make sure that the loop ends ?
What algorithm is implemented here ?

Reply:
It's a modification of the approach listed in the original issue: https://github.com/nilearn/nilearn/issues/3711#issue-1675696871, with the modifications aiming to ensure that the contour does not cross itself. I'm not sure how it could fail to end, but if preferred I think I could update it to a for loop.

Reply:
I'd more comfortable with a for loop indeed...

Reply:
Sounds good! I cleaned up the sorting procedure a bit and added a few more explanatory comments cb8ad06.